pyrobosim/pyrobosim/core/robot.py

```diff
@@ -84,6 +83,15 @@ def __init__(
         if name == "world":
             raise ValueError("Robots cannot be named 'world'.")
 
+        # Logger for this robot
+        self.logger = logging.getLogger(self.name)
+        self.logger.setLevel(logging.INFO)
+        log_formatter = logging.Formatter("[%(name)s] %(levelname)s: %(message)s")
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(log_formatter)
+        self.logger.addHandler(console_handler)
+        self.logger.propagate = True
```

Comment:
I think it might be worth moving this into a `get_logger(name=...)` function in the global space. Could be handy to have everything log related up there, and it would make it much easier to change the configuration, whether it be adding file logging or changing the log level. The global logger could be constructed from it, too.

Reply:
nice idea -- I like how this turned out!